kratos/geometries/tetrahedra_3d_4.h

```diff
@@ -1410,6 +1418,69 @@ template<class TPointType> class Tetrahedra3D4 : public Geometry<TPointType>
                 rResult[i] = DN_DX;
     }
 
+    /**
+     * returns the second order derivatives of all shape functions
+     * in given arbitrary points
+     * @param rResult a third order tensor which contains the second derivatives
+     * @param rPoint the given point the second order derivatives are calculated in
+     */
+    ShapeFunctionsSecondDerivativesType& ShapeFunctionsSecondDerivatives( ShapeFunctionsSecondDerivativesType& rResult, const CoordinatesArrayType& rPoint ) const override
+    {
+        if ( rResult.size() != this->PointsNumber() )
+        {
+            // KLUDGE: While there is a bug in ublas vector resize, I have to put this beside resizing!!
+            ShapeFunctionsGradientsType temp( this->PointsNumber() );
+            rResult.swap( temp );
+        }
+
+        for ( unsigned int i = 0; i < this->PointsNumber(); i++ )
+        {
+            rResult[i].resize( 3, 3, false );
```

Comment:
```suggestion
            rResult[i] = ZeroMatrix( 3, 3);
```
(and remove all the zeros below).